Test/default.cpp

```diff
@@ -8,47 +8,108 @@
 // C++ libraries
 
 // Other libraries
+#include <Arduino.h>
 
 // NS2 headers
- 
- // I think I'm writing this as Arduino code not c++? 
-int aRegCurr = 17;
-int dRegCurr = 18;
-int dRegPG = 20; 
 
-void loop()
+
+/* - - - - - - Initialization - - - - - - */
+// pin numbers
+const int PIN_AREG_CURR = 17;   // analog regulator current pin
+const int PIN_DREG_CURR = 18;   // digital regulator current pin
+const int PIN_DREG_PG = 20;     // digital regulator 'power good' pin
+const int PIN_WD_RESET = 2;     // watchdog reset pin
+
+// Acceptable voltage range
+const float VMIN = 3.25;        // minimum accpetable voltage
+const float VMAX = 3.35;        // maximum acceptable voltage
+
+// timing
+const int SAMPLE_INTERVAL = 3000;       // time between samples in ms
+const int WD_RESET_INTERVAL = 1000;     // watchdog feeding interval, ms
+const int WD_PULSE_DUR = 10;            // watchdog reset signal duration, MICROSECONDS!!!
+
+int wdLastFeedMillis;
+int lastSampleMillis;
+
+// conversion factors
+const int ADC_BINS = 1024;    // number of ADC bins
+const float HI_VOLTAGE = 3.3; // high board voltage, 3.3V for teensy
+const float R_SENSE = 0.02;    // current sensor resistance, ohms
+
+
+/* - - - - - - Functions - - - - - - */
+
+/* - - - - - printStatus - - - - - */
+void printStatus()
+{
+  float aCurr = HI_VOLTAGE*(float)analogRead(PIN_AREG_CURR)/(ADC_BINS*R_SENSE);   // analog regulator current, Amps
```

Comment:
Also, the current (both analog and digital) is actually the same as the voltage reading of the pin, because the voltage is in units of 1 V/A. I did the math to show Jackson, but I can also show you later Jaret. I'll manually determine any error in the lab.